### DISCUSSION TITLE
Logging to one of multiple loggers

### DISCUSSION THREAD

**Discussion by jbuckman** [Lightning-AI/pytorch-lightning#7611] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi, I'm confused as to the intended syntax for using multiple loggers. Before, I used:

```self.log(k, v, sync_dists=True)```

Now, I want to instead log to only one of my three loggers. I'd like to do something like this:

```self.log(k, v, sync_dists=True, logger_id=2```

[This page](https://pytorch-lightning.readthedocs.io/en/stable/common/loggers.html#multiple-loggers) does not give a direct way to do this. Instead, it makes me directly access the loggers themselves. I can call the individual write functions on my loggers, but I'm concerned that I will be losing whatever useful functionality comes with sync_dists, and my code will break in the DDP case.

How should I be doing this?

**Comment by carmocca** [comment]
Hello!

`sync_dist` is to synchronize the logged values in a distributed setting, it has nothing to do with logging.

There is no way to set what values go into which logger when you are calling `self.log(...)`, you would have to do the filtering in the logger itself.
